File: Ejercicio7.py

def evaluar_empleado():
    try:
        puntuacion = int(input("Ingresa la puntuación del empleado (0-100): "))

        if puntuacion >= 90:
            nivel = "Excelente"
            recompensa = 1000
        elif puntuacion >= 70:
            nivel = "Bueno"
            recompensa = 500
        elif puntuacion >= 50:
            nivel = "Regular"
            recompensa = 100
        else:
            nivel = "Insuficiente"
            recompensa = 0

        print(f"Nivel de rendimiento: {nivel}")
        print(f"Recompensa monetaria: ${recompensa}")
    except ValueError:
        print("Entrada inválida. Por favor, ingresa un número entero para la puntuación.")

if __name__ == "__main__":
    evaluar_empleado()

File: Ejercicio7.hs
module Ejercicio7 where

import System.IO


evaluarEmpleado :: Int -> (String, Double)
evaluarEmpleado puntuacion
    | puntuacion >= 90 = ("Excelente", 1000.0)
    | puntuacion >= 70 = ("Bueno", 500.0)
    | puntuacion >= 50 = ("Regular", 100.0)
    | otherwise        = ("Insuficiente", 0.0)

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering
    putStr "Ingrese la puntuación del empleado (0-100): "
    puntuacionStr <- getLine
    let puntuacion = read puntuacionStr :: Int
    let (nivel, recompensa) = evaluarEmpleado puntuacion
    putStrLn $ "Nivel de rendimiento: " ++ nivel
    putStrLn $ "Recompensa monetaria: " ++ show recompensa


